astropy/stats/lombscargle/_lombscargle.py

```diff
@@ -0,0 +1,256 @@
+        maximum_frequency argument.
+
+        Parameters
+        ----------
+        samples_per_peak : float (optional, default=5)
+            The approximate number of desired samples across the typical peak
+        nyquist_factor : float (optional, default=5)
+            The multiple of the average nyquist frequency used to choose the
+            maximum frequency if maximum_frequency is not provided.
+        minimum_frequency : float (optional)
+            If specified, then use this minimum frequency rather than one
+            chosen based on the size of the baseline.
+        maximum_frequency : float (optional)
+            If specified, then use this maximum frequency rather than one
+            chosen based on the average nyquist frequency.
+
+        Returns
+        -------
+        frequency : ndarray or Quantity
+            The heuristically-determined optimal frequency bin
+        """
+        t = np.asanyarray(self.t)
+        baseline = t.max() - t.min()
+        n_samples = t.size
+
+        df = 1. / baseline / samples_per_peak
+
+        if minimum_frequency is not None:
+            f0 = minimum_frequency
+        else:
+            f0 = 0.5 * df
+
+        if maximum_frequency is not None:
+            Nf = int(np.ceil((maximum_frequency - f0) / df))
+        else:
+            Nf = int(0.5 * samples_per_peak * nyquist_factor * n_samples)
+
+        return f0 + df * np.arange(Nf)
+
+    def autopower(self, method='auto', method_kwds=None,
+                  normalization='normalized', **kwargs):
+        """Compute Lomb-Scargle power at automatically-determined frequencies
+
+        Parameters
+        ----------
+        method : string (optional)
+            specify the lomb scargle implementation to use. Options are:
+
+            - 'auto': choose the best method based on the input
+            - 'fast': use the O[N log N] fast method. Note that this requires
+              evenly-spaced frequencies: by default this will be checked unless
+              ``assume_regular_frequency`` is set to True.
+            - 'slow': use the O[N^2] pure-python implementation
+            - 'cython': use the O[N^2] cython implementation. This is slightly
+              faster than method='slow', but much more memory efficient.
+            - 'chi2': use the O[N^2] chi2/linear-fitting implementation
+            - 'fastchi2': use the O[N log N] chi2 implementation. Note that this
+              requires evenly-spaced frequencies: by default this will be checked
+              unless ``assume_regular_frequency`` is set to True.
+            - 'scipy': use ``scipy.signal.lombscargle``, which is an O[N^2]
+              implementation written in C. Note that this does not support
+              heteroskedastic errors.
+
+        method_kwds : dict (optional)
+            additional keywords to pass to the lomb-scargle method
+        normalization : string (optional, default='normalized')
```

Comment:
There are a *lot* of different normalizations of the periodogram used in the literature. I was going to dig through and implement others, but I didn't have the time. I thought I'd leave it as a string in case people want to add more in the future.